src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -0,0 +1,206 @@
+package resources
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	redpandav1alpha1 "github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const fsGroup = 101
+
+type StatefulSetResource struct {
+	client.Client
+	scheme		*runtime.Scheme
+	pandaCluster	*redpandav1alpha1.Cluster
+
+	LastObservedState	*appsv1.StatefulSet
```

Comment:
Are you introducing `LastObservedState` for future changes, such as updates after creation? It doesn't seem to be needed in this commit, so I wanted to clarify.

Reply:
it's needed for status computation in cluster_controller